controllers/cloudstackfailuredomain_controller.go

```diff
@@ -127,7 +127,7 @@ func (r *CloudStackFailureDomainReconciliationRunner) ReconcileDelete() (ctrl.Re
 // and requeus until all CloudStack machines are cleared from the FailureDomain.
 func (r *CloudStackFailureDomainReconciliationRunner) ClearMachines() (ctrl.Result, error) {
 	machines := &infrav1.CloudStackMachineList{}
-	if err := r.K8sClient.List(r.RequestCtx, machines, client.MatchingLabels{infrav1.FailureDomainLabelName: r.ReconciliationSubject.Name}); err != nil {
+	if err := r.K8sClient.List(r.RequestCtx, machines, client.MatchingLabels{infrav1.FailureDomainLabelName: r.ReconciliationSubject.Spec.Name}); err != nil {
```

Comment:
why is the failuredomain's name in its spec? Is this because of the hashing changes?

Reply:
Yes. The spec has the real name and the meta name is the hash.